frappe/desk/doctype/todo/todo.py

```diff
@@ -97,9 +98,22 @@ def has_permission(doc, user):
 	else:
 		return doc.owner==user or doc.assigned_by==user
 
+def update_overdue():
+	today = nowdate()
+	todos = frappe.db.get_all('ToDo', filters={'status': 'Open'}, fields=['name', 'date'])
```

Comment:
please filter by date here! you are scanning twice!

Reply:
Once a todo is Overdue, It shouldn't be scanned, since the filter is on status "Open".

If we filter by date, older todos, which are overdue, will be ignored.

Reply:
I mean the query is already scanning the todo table, you can add it as a filter `('date': ['<', nowdate()])`